src/splitRegistry.js

```diff
@@ -1,41 +1,38 @@
-var SplitRegistry = function(splitArray) {
-  this._splitArray = splitArray;
-  this._loaded = splitArray !== null;
-};
-
-SplitRegistry.prototype.getSplit = function(splitName) {
-  return this.getSplits()[splitName];
-};
-
-SplitRegistry.prototype.isLoaded = function() {
-  return this._loaded;
-};
-
-SplitRegistry.prototype.asV1Hash = function() {
-  var v1Hash = {};
-  for (var splitName in this.getSplits()) {
-    var split = this._splits[splitName];
-    v1Hash[splitName] = split.getWeighting();
+class SplitRegistry {
+  constructor(splitArray) {
+    this._splitArray = splitArray;
+    this._loaded = splitArray !== null;
   }
 
-  return v1Hash;
-};
+  getSplit(splitName) {
+    return this.getSplits()[splitName];
+  }
 
-SplitRegistry.prototype.getSplits = function() {
-  if (!this._loaded) {
-    return {};
+  isLoaded() {
+    return this._loaded;
   }
 
-  if (!this._splits) {
-    this._splits = {};
-    this._splitArray.forEach(
-      function(split) {
-        this._splits[split.getName()] = split;
-      }.bind(this)
-    );
+  asV1Hash() {
+    const v1Hash = {};
+    for (let splitName in this.getSplits()) {
+      const split = this._splits[splitName];
+      v1Hash[splitName] = split.getWeighting();
+    }
+
+    return v1Hash;
   }
 
-  return this._splits;
-};
+  getSplits() {
+    if (!this._loaded) {
+      return {};
+    }
+
+    if (!this._splits) {
+      this._splits = this._splitArray.reduce((result, split) => ({ ...result, [split.getName()]: split }), {});
```

Comment:
Does this create a new accumulator object on each iteration, would `Object.assign(result, {[split.getName()]: split})` work better or even `new Map(this.splitArray.map(split => [split.getName(), split]))`?

Reply:
> Does this create a new accumulator object on each iteration

Yes, since it is returning a new object with the properties of the existing `result` accumulator and the new split property added. 

`Object.assign` would technically prevent creating a new object in each iteration because it would mutate the `result` object with the new property.

A `Map` would be an entirely new interface for the `_splits`, which would be nicer for internal usage, but should be held off for a follow up refactor IMO.

Reply:
Definitely agree on switching to using a `Map`.